pkg/machines/components/nicEdit.jsx

```diff
@@ -53,10 +53,26 @@ class EditNICModal extends React.Component {
     constructor(props) {
         super(props);
 
+        let defaultNetworkSource;
+        let currentSource;
+        let availableSources = [];
+        if (props.network.type === "network") {
+            availableSources = props.networks.map(network => network.name);
+            currentSource = props.network.source.network;
+        } else if (props.network.type === "direct") {
+            availableSources = props.nodeDevices;
```

Comment:
props.nodeDevices is not a array of strings, but array of objects, so you need to get and nodeDevice names from those objects (similar to what you do with networks a few lines up.

Reply:
Thanks, done.